core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -2518,12 +2592,32 @@ public UfsInfo getUfsInfo(long mountId) {
    * Loads metadata for the path if it is (non-existing || load direct children is set).
    *
    * @param rpcContext the rpc context
-   * @param inodePath the {@link LockedInodePath} to load the metadata for
    * @param context the load metadata context
    */
-  private void loadMetadataIfNotExist(RpcContext rpcContext, LockedInodePath inodePath,
-      LoadMetadataContext context, boolean isGetFileInfo) {
-    Preconditions.checkState(inodePath.getLockPattern() == LockPattern.READ);
+  private void loadMetadataIfNotExist(RpcContext rpcContext, AlluxioURI path,
+      LoadMetadataContext context, boolean isGetFileInfo)
+      throws InvalidPathException, AccessControlException {
+    LockingScheme scheme = new LockingScheme(path, LockPattern.READ, false);
+    boolean lm;
+    try (LockedInodePath inodePath = mInodeTree.lockInodePath(scheme)) {
+      lm = shouldLoadMetadataIfNotExists(inodePath, context);
```

Comment:
the rpcs use `shouldLoadMetadataIfNotExists` to check to load metadata, but this method also calls `shouldLoadMetadataIfNotExists` again. Is that intentional, and why do we need to check again?

Reply:
It was originally intentional because I was trying to preserve the old behavior, and it's possible that in between the original check and running this function the tree changed, so this can prevent us from loading the metadata unnecessarily

But I am thinking more about it now; the cost of the lock to perform this check vs the probability of the path being loaded since the last check outside this function is likely low, and so the cost of performing this lock vs the amount of CPU time it _could_ save is likely going to do more harm than good. I'll remove it.